src/components/Popover/Popover.jsx

```diff
@@ -52,3 +56,32 @@ Content.propTypes = {
 };
 
 Popover.Content = Content;
+
+/**
+ * This is to ensure lazy mounting which will not mount the content until the tippy is shown.
+ *
+ * By default, Tippy mounts your content or render elements into a container element once created,
+ * even if the tippy isn't mounted on the DOM. In most cases, this is fine, but in performance-sensitive scenarios
+ * or cases where mounting the component should fire effects only when the tippy mounted, you can lazify the component.
+ */
+const LazyTippy = React.forwardRef((props, ref) => {
```

Comment:
It may be worth adding an optional prop and logic to the lazy popover component to show something while it's rendering. It can be `null` by default and we will show nothing, but you can pass a loader or skeleton to display during that heavy popover rendering. In the quick search it was done like this - https://github.com/xola/x2-seller/blob/development/apps/seller/src/components/PurchaseModification/CheckInPurchaseGuests/CheckInPurchaseGuestList.tsx#L21-L30

Reply:
we can check this to display fallback while rendering https://react.dev/reference/react/Suspense

Reply:
I'll check this. React 17 has Suspense but I don't think it's full stable. I'll check, and see if I can use a work around